src/pygeosx/pyhdf5/PyHDF5.cpp

```diff
@@ -0,0 +1,294 @@
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
+// Source includes
+#include "../pygeosx.hpp"
+#include "PyHDF5.hpp"
+
+#include "fileIO/timeHistory/TimeHistoryCollection.hpp"
+
+
+#define VERIFY_NON_NULL_SELF( self ) \
+  PYTHON_ERROR_IF( self == nullptr, PyExc_RuntimeError, "Passed a nullptr as self.", nullptr )
+
+#define VERIFY_INITIALIZED( self ) \
+  PYTHON_ERROR_IF( self->collection.size() == 0, PyExc_RuntimeError, "The PyHDF5 is not initialized.", nullptr )
+
+
+namespace geosx
+{
+extern std::unique_ptr< GeosxState > g_state;
+
+extern bool g_alreadyInitialized;
+
+namespace python
+{
+
+struct PyHDF5
```

Comment:
I'd also rename this to something related to time history and not HDF5.

Reply:
I renamed it as `PyHistory`